types/signature.go

```diff
@@ -50,13 +50,29 @@ type LogicSig struct {
 	// OR hashed to be the Address of an account.
 	Logic []byte `codec:"l"`
 
-	Sig  Signature   `codec:"sig"`
+	// The signature of the account that has delegated to this LogicSig, if any
+	Sig Signature `codec:"sig"`
+	// The key that provided Sig, if any
+	SigKey ed25519.PublicKey `codec:"sigkey"`
```

Comment:
imo adding a new field into a struct that is going to be serialized, and removing it before serializing is error-prone. Please consider alternative approach - wrapping this canonical struct or passing it along with `LogicSig` type